### DISCUSSION TITLE
How to read image from URL to crop it?

### DISCUSSION THREAD

**Discussion by glslmn** [ImageMagick/ImageMagick#1924] [General]
Hi everyone, 

I have many images remote and I want to crop an save them into my local machine with  ImageMagick. I am on Python 3.7 hence run subrocess module for my script.  The following script works just as I want if the source image is already saved in my local server

```
subprocess.call([
        'sudo',
        'convert',
        ‘path_to_the_source_folder/source_file.png’,
        '-fuzz',
        '10%%',
        '-trim',
        ‘path_to_the_target_folder/target_file.png’]) 
```

but since I have many images, I don’t want to first save all of them into my local server and then crop them. Is there any way which I can read the images from URL and then crop and save them with ImageMagick?

Thank you!

**Comment by snibgo** [comment]
You can put the URL in the place of the source filename.

**Reply by glslmn** [reply]
when I pass `source URL` in the place of  ‘path_to_the_source_folder/source_file.png’ unfortunatelly. It doesn't generate the cropped image in my target folder

**Comment by snibgo** [comment]
What error do you get? Do you even trap and report errors?

Perhaps your policy.xml doesn't permit downloads.

**Reply by glslmn** [reply]
Yes I should have mentioned about the output. So when I run the command it outputs 1. I am running on jupyter notebook.
I dont know about my policy.xml, I am an absolute newbie for ImageMagick, so if it is so, how can I change it?

**Comment by fmw42** [comment]
What is your Imagemagick version?  Can you open a terminal window and type

`convert -version`

or

`magick -version`

and let us know what you get. Or put those in Python subprocess command.

What is the URL you want to download? Is it http: or https:

**Reply by glslmn** [reply]
the version is: ImageMagick 7.0.10-7 Q16 x86_64 2020-04-21

and the URL type is https:

**Reply by fmw42** [reply]
Please show the results of

`magick -version`

Try the following and see if you get an image 

`magick https://imagemagick.org/image/wizard.png wizard_downloaded.gif`

**Reply by glslmn** [reply]
when I run the command on terminal window, the image is downloaded. Wheres when I run the command on jupyter notebook with 
os.system('magick https://imagemagick.org/image/wizard.png wizard_downloaded.gif'), it outputs "32512" and no image

**Reply by glslmn** [reply]
```
magick -version
``` 
outputs

Version: ImageMagick 7.0.10-7 Q16 x86_64 2020-04-21 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): bzlib fontconfig freetype jng jpeg lzma pangocairo png x zlib

**Reply by fmw42** [reply]
Sorry, I know little about Jupyter. You will need to check the policy.xml file on Imagemagick in your Jupyter notebook. You may need to give read|write permission for https. See https://imagemagick.org/source/policy.xml

**Comment by snibgo** [comment]
> Yes I should have mentioned about the output. So when I run the command it outputs 1. I am running on jupyter notebook.
>  it outputs "32512" and no image

But what error message does ImageMagick send to stderr?

Try "magick -list policy". What is the stdout text output?

**Reply by glslmn** [reply]
It didnt output any error. I think it just fails silently
when I type
 ```
 magick -list policy
```
it outputs
```
Path: [built-in]
  Policy: Undefined
    rights: None 
```

**Comment by snibgo** [comment]
Okay, good, so it isn't an ImageMagick problem.

Perhaps jupyter notebook doesn't allow programs to download files.

**Reply by glslmn** [reply]
Yes, it works perfectly well on shell. I am using jupyter on a cluster so probably it has something to with that
